p2p/src/net/default_backend/mod.rs

```diff
@@ -202,19 +211,12 @@ impl<T: TransportSocket> MessagingService for MessagingHandle<T> {
     }
 
     fn make_announcement(&mut self, announcement: Announcement) -> crate::Result<()> {
-        let message = announcement.encode();
-        if message.len() > ANNOUNCEMENT_MAX_SIZE {
```

Comment:
Why remove this check? It doesn't hurt. It can be an assert at least.

Reply:
The check will be performed in `EncoderDecoder` anyway, I'm not sure if it makes sense to perform the same check on several layers.